scripts/scil_summarize_bundles_similarity_measures.py

```diff
@@ -0,0 +1,328 @@
+    indices = np.where(density_1 + density_2 > 0)
+    indices_endpoints = np.where(endpoints_density_1 + endpoints_density_2 > 0)
+    dice_vox_endpoints, w_dice_vox_endpoints = compute_dice_voxel(
+        endpoints_density_1,
+        endpoints_density_2)
+    density_correlation = np.corrcoef(
+        density_1[indices], density_2[indices])[0, 1]
+    corrcoef = np.corrcoef(endpoints_density_1[indices_endpoints],
+                           endpoints_density_2[indices_endpoints])
+    density_correlation_endpoints = corrcoef[0, 1]
+
+    measures_name = ['bundle_adjacency_voxels',
+                     'dice_voxels', 'w_dice_voxels',
+                     'volume_overlap',
+                     'volume_overreach',
+                     'dice_voxels_endpoints',
+                     'w_dice_voxels_endpoints',
+                     'volume_overlap_endpoints',
+                     'volume_overreach_endpoints',
+                     'density_correlation',
+                     'density_correlation_endpoints']
+    measures = [bundle_adjacency_voxel,
+                dice_vox, w_dice_vox,
+                volume_overlap * voxel_size,
+                volume_overreach * voxel_size,
+                dice_vox_endpoints,
+                w_dice_vox_endpoints,
+                volume_overlap_endpoints * voxel_size,
+                volume_overreach_endpoints * voxel_size,
+                density_correlation,
+                density_correlation_endpoints]
+
+    if not disable_streamline_distance:
+        measures_name += ['bundle_adjacency_streamlines']
+        measures += [bundle_adjacency_streamlines]
+
+    # Only when the tractograms are exactly the same
+    if same_tractogram:
+        dice_streamlines, streamlines_intersect, streamlines_union = \
+            compute_dice_streamlines(bundle_1, bundle_2)
+        streamlines_count_overlap = len(streamlines_intersect)
+        streamlines_count_overreach = len(
+            streamlines_union) - len(streamlines_intersect)
+        measures_name += ['dice_streamlines',
+                          'streamlines_count_overlap',
+                          'streamlines_count_overreach']
+        measures += [dice_streamlines,
+                     streamlines_count_overlap,
+                     streamlines_count_overreach]
+
+    return dict(zip(measures_name, measures))
+
+
+def main():
+    parser = _build_args_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, args.in_bundles)
+    assert_outputs_exist(parser, args, [args.out_json])
+
+    nbr_cpu = args.processes if args.processes else multiprocessing.cpu_count()
+    if nbr_cpu <= 0:
+        parser.error('Number of processes cannot be <= 0.')
+    elif nbr_cpu > multiprocessing.cpu_count():
+        parser.error('Max number of processes is {}. Got {}.'.format(
+            multiprocessing.cpu_count(), nbr_cpu))
+
+    if not os.path.isdir('tmp_measures/'):
+        os.mkdir('tmp_measures/')
```

Comment:
should we delete everything otherwise?

Reply:
If it exists, it could be for a re-run, which could be drastically faster. Also because it is base on an hashing naming convention, there is no conflict (and if files exist it will speed up everything)

I would leave the deleting to the user, the -f should, in my opinion, only target the output.

Reply:
:+1: